src/ExpressionEvaluator/CSharp/Source/ExpressionCompiler/EvaluationContext.cs

```diff
@@ -222,14 +222,15 @@ internal CompilationContext CreateCompilationContext()
             out ImmutableArray<int> methodTokens,
             out ImmutableArray<string> errorMessages)
         {
+            var context = CreateCompilationContext();
+            bool isInFieldKeywordContext = context.IsInFieldKeywordContext();
             var diagnostics = DiagnosticBag.GetInstance();
-            var syntaxNodes = expressions.SelectAsArray(expr => Parse(expr, treatAsExpression: true, diagnostics, out var formatSpecifiers));
+            var syntaxNodes = expressions.SelectAsArray((expr, isInFieldKeywordContext) => Parse(expr, isInFieldKeywordContext, treatAsExpression: true, diagnostics, out var formatSpecifiers), isInFieldKeywordContext);
```

Comment:
It looks like the `SelectAsArray` overload which takes an extra argument, is being used to avoid a delegate allocation, but, it looks like a delegate allocation is already happening due to capture of `diagnostics`. Consider passing `(isInFieldKeywordContext, diagnostics)` as the extra argument to avoid this.